frontend/src/components/Forms/parties/AddRolesForm.js

```diff
@@ -0,0 +1,126 @@
+import CustomPropTypes from "@/customPropTypes";
+import * as FORM from "@/constants/forms";
+import { renderConfig } from "@/components/common/config";
+import { required } from "@/utils/Validate";
+import { COLOR } from "@/constants/styles";
+
+const { layoutGrey } = COLOR;
+
+const propTypes = {
+  addField: PropTypes.func.isRequired,
+  removeField: PropTypes.func.isRequired,
+  handleChange: PropTypes.func.isRequired,
+  handleSelect: PropTypes.func.isRequired,
+  roleNumbers: PropTypes.arrayOf(PropTypes.string).isRequired,
+  partyRelationshipTypesList: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+  mineNameList: PropTypes.arrayOf(CustomPropTypes.mineName).isRequired,
+};
+
+const complexRelationships = ["EOR", "PMT"];
+const simpleRelationships = (typesList) =>
+  typesList.filter(({ value }) => !complexRelationships.includes(value));
+
+const panelHeader = (removeField, roleNumber) => (
+  <div className="inline-flex between">
+    <Form.Item style={{ marginTop: "15px" }} label={`Role ${roleNumber}`} />
+    <div>
+      <Popconfirm
+        placement="topRight"
+        title={`Are you sure you want to remove Role ${roleNumber}?`}
+        onConfirm={removeField(roleNumber)}
+        okText="Yes"
+        cancelText="No"
+      >
+        <Button ghost>
+          <Icon type="minus-circle" theme="outlined" style={{ color: "#BC2929" }} />
+        </Button>
+      </Popconfirm>
+    </div>
+  </div>
+);
+
+const transformMineNames = (names) =>
+  names.map(({ mine_name, guid }) => (
+    <AutoComplete.Option key={guid} value={guid}>
+      {mine_name}
+    </AutoComplete.Option>
+  ));
+
+export const AddRolesForm = (props) => (
+  <div>
+    <Form>
+      <Collapse>
+        {props.roleNumbers.map((roleNumber) => (
+          <Collapse.Panel header={panelHeader(props.removeField, roleNumber)} key={roleNumber}>
+            <Row gutter={16}>
+              <Col span={12}>
+                <Field
+                  label="Role *"
+                  id={`mine_party_appt_type_code-${roleNumber}`}
+                  name={`mine_party_appt_type_code-${roleNumber}`}
+                  placeholder="Search for Mine"
+                  component={renderConfig.SELECT}
+                  data={simpleRelationships(props.partyRelationshipTypesList)}
+                  validate={[required]}
+                />
+              </Col>
+              <Col span={12}>
+                <Form.Item label="Mine *">
+                  <Field
+                    id={`mine_guid-${roleNumber}`}
+                    name={`mine_guid-${roleNumber}`}
+                    component={renderConfig.AUTOCOMPLETE}
+                    placeholder="Please add Mine"
+                    data={transformMineNames(props.mineNameList)}
+                    handleChange={props.handleChange(roleNumber)}
+                    handleSelect={props.handleSelect(roleNumber)}
+                    iconColor="rgba(0, 0, 0, 0.25)"
```

Comment:
Is there a reason why this is rgba and not hex? 
Also can this be stored in a variable with the rest of the colors?

Reply:
1) I just pulled it directly from Ant Design (it's used to match their icon colour in an input field). I can likely get it to be a HEX -- not an expert on how that works with transparency, though.
2) Sure!